libsupport/include/katana/OpaqueID.h

```diff
@@ -216,6 +237,16 @@ struct OpaqueIDLinear : public OpaqueIDOrderedWithValue<_IDType, _Value> {
   DifferenceType operator-(_IDType v) const {
     return DifferenceType(this->value()) - DifferenceType(v.value());
   }
+
+  /// Be careful! std::numeric_limits is only specialized for a handful of
+  /// numeric types and the default behavior of std::numeric_limits<T> is to
+  /// return T().
+  ///
+  /// So if your _Value is not a language-provided numeric type
+  /// you should specialize std::numeric_limits<_Value>, specialize this
+  /// sentinel function, or be prepared to use _Value as your sentinel value
+  static constexpr _IDType sentinel =
+      _IDType(std::numeric_limits<_Value>::max());
```

Comment:
Use `boost::math::tools::max_value`. That will fail instead of producing 0 in cases where it is not specialized. It will work for anything that specializes either that function or `numeric_limits`.

If subclasses override this, will it prevent static errors? If not, maybe this needs an `enable_if` template of some kind to allow using `OpaqueIDLinear` on types without `numeric_limits`. Though I'm not sure there is any reason to support that.

Reply:
I think we could safely wait and find out